Spotify-ObjC/Spotify-ObjC/SASearchViewController.m

```diff
@@ -42,32 +37,40 @@ - (void)setupVC {
     self.textField.backgroundColor = backgroundColor;
     self.navigationController.navigationBar.barTintColor = backgroundColor;
     self.navigationController.navigationBar.backgroundColor = backgroundColor;
-    _textField.attributedPlaceholder = [[NSAttributedString alloc] initWithString:@"Search Spotify" attributes:@{NSForegroundColorAttributeName: placeholderTextColor}];
+    self.textField.attributedPlaceholder = [[NSAttributedString alloc] initWithString:@"Search Spotify" attributes:@{NSForegroundColorAttributeName: placeholderTextColor}];
 }
 
 //MARK: - UITextFieldDelegate
 
 - (BOOL)textFieldShouldReturn:(UITextField *)textField {
-    
     NSString *searchQuery = self.textField.text;
-    
     if ([searchQuery isEqualToString:@""]) {
         return YES;
     } else {
-        // Run our query
+        [requestManager getArtistsWithQuery:searchQuery completion:^(SAResponse *response) {
+            if (*(response.response) == Failure) {
+                NSLog(@"Error fetching artists");
+            } else {
+                NSLog(@"Success!!!");
+                artists = response.artists;
+                [self.tableView reloadData];
+            }
+        }];
     }
     return YES;
 }
 
 // MARK: - UITableViewDataSource
 
 - (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
-    return 10;
+    return [artists count];
 }
 
 - (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(nonnull NSIndexPath *)indexPath {
     UITableViewCell *searchCell = [tableView dequeueReusableCellWithIdentifier: cellIdentifier];
-    searchCell.textLabel.text = @"Zedd";
+    SAArtist *currentArtist = [[SAArtist alloc] init];
```

Comment:
*nit*: You should probably either be declaring it without init-ing it, or directly initializing it from the array. No reason to make one just to set it to something different. 

If you are worried about if its going to be an artist or not, we should check it that it is with an if and set the cell text within that